File: core/src/main/java/core/rag/Opinion.java
package core.rag;

public interface Opinion {
    String enrollOpinion(String userId, String questionId, String title, String contents);
    String updateOpinion(String userId, String opinionId, String title, String contents);
    String deleteOpinion(String userId, String opinionId);
    String readOpinions(String questionId, int startNum, int endNum);
}

File: core/src/main/kotlin/core/rag/Opinion.kt
package core.rag

import java.time.LocalDateTime

data class Opinion (
    val questionId: String,
    val title: String,
    val content: String,
    val userId: String,
    val opinionId: String? = null,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null
)

